lib/spack/spack/environment.py

```diff
@@ -1209,14 +1210,24 @@ def _concretize_separately(self, tests=False):
                 self._add_concrete_spec(s, concrete, new=False)
 
         # Concretize any new user specs that we haven't concretized yet
-        concretized_specs = []
+        arguments, root_specs = [], []
         for uspec, uspec_constraints in zip(
-                self.user_specs, self.user_specs.specs_as_constraints):
+                self.user_specs, self.user_specs.specs_as_constraints
+        ):
             if uspec not in old_concretized_user_specs:
-                concrete = _concretize_from_constraints(uspec_constraints, tests=tests)
-                self._add_concrete_spec(uspec, concrete)
-                concretized_specs.append((uspec, concrete))
-        return concretized_specs
+                root_specs.append(uspec)
+                arguments.append((uspec_constraints, tests))
+
+        # Solve the environment in parallel
+        pool = multiprocessing.pool.Pool()
+        concretized_root_specs = pool.starmap(_concretize_from_constraints, arguments)
+
```

Comment:
@alalazo This is more of a question.  Do you have some guidance when we should `terminate()` and `join()` the threadpool, and when it's not necessary or not desirable.  In other examples in spack (`compilers/__init__.py`, `util/web.py`, `relocate.py`) I see we either `close()` or `terminate()` and `join()`.

Reply:
That's a good comment. I forgot to call those cleanup functions here. Reading the standard library, it seems that the default behavior in Python is to `terminate` before `join`, but I called `close` since in our case it should be equivalent.